expertsystem/state/particle.py

```diff
@@ -526,27 +529,32 @@ def initialize_graph(
     return new_graphs
 
 
-def __safe_set_spin_projections(state: StateDefinition) -> StateWithSpins:
-    if isinstance(state, str):
-        particle_name = state
-        particle = DATABASE[state]
-        spin_projections = arange(  # type: ignore
-            -particle.state.spin, particle.state.spin + 1, 1.0
-        ).tolist()
-        if particle.mass == 0.0:
-            if 0.0 in spin_projections:
-                del spin_projections[spin_projections.index(0.0)]
-        state = (particle_name, spin_projections)
-    return state
+def __safe_set_spin_projections(
+    list_of_states: List[StateDefinition],
+) -> List[StateWithSpins]:
+    def safe_set_spin_projections(state: StateDefinition) -> StateWithSpins:
+        if isinstance(state, str):
+            particle_name = state
+            particle = DATABASE[state]
+            spin_projections = arange(  # type: ignore
+                -particle.state.spin, particle.state.spin + 1, 1.0
+            ).tolist()
+            if particle.mass == 0.0:
+                if 0.0 in spin_projections:
+                    del spin_projections[spin_projections.index(0.0)]
+            state = (particle_name, spin_projections)
+        return state
+
+    return [safe_set_spin_projections(state) for state in list_of_states]
```

Comment:
Can you please avoid to do such unrelated refactoring work, if it is being overhauls by upcoming issues anyway

Reply:
Kicked out